### ISSUE TITLE
Opinion for Fast PatchMerging Layer

### ISSUE THREAD

**Issue by developer0hye** [svip-lab/AS-MLP#5]
https://github.com/svip-lab/AS-MLP/blob/946848d957e5b1c33226ddacaa35f422a103ddfb/models/as_mlp.py#L187-L222

Thanks for sharing your work!

Why did you design PathMerging Layer normalizing features first?

I think that if we normalize features after reduce layer, we can reduce the size of model and flops like below code.

Did you do ablation study for this layer?

```python
 class PatchMerging(nn.Module): 
     r""" Patch Merging Layer. 
  
     Args: 
         input_resolution (tuple[int]): Resolution of input feature. 
         dim (int): Number of input channels. 
         norm_layer (nn.Module, optional): Normalization layer.  Default: nn.LayerNorm 
     """ 
  
     def __init__(self, input_resolution, dim, norm_layer=nn.LayerNorm): 
         super().__init__() 
         self.input_resolution = input_resolution 
         self.dim = dim 
         self.reduction = nn.Conv2d(4 * dim, 2 * dim, 1, 1, bias=False) 
         self.norm = norm_layer(2 * dim) 
  
     def forward(self, x): 
         """ 
         x: B, H*W, C 
         """ 
         B, C, H, W = x.shape 
         #assert L == H * W, "input feature has wrong size" 
         assert H % 2 == 0 and W % 2 == 0, f"x size ({H}*{W}) are not even." 
  
         x = x.view(B, C, H, W) 
  
         x0 = x[:, :, 0::2, 0::2]  # B C H/2 W/2  
         x1 = x[:, :, 1::2, 0::2]  # B C H/2 W/2  
         x2 = x[:, :, 0::2, 1::2]  # B C H/2 W/2  
         x3 = x[:, :, 1::2, 1::2]  # B C H/2 W/2  
         x = torch.cat([x0, x1, x2, x3], 1)  # B 4*C H/2 W/2  

         x = self.reduction(x) 
         x = self.norm(x) 
         return x 
```

**Comment by niujinshuchong** [comment]
@developer0hye We didn't try this. The PatchMerging we used is from Swin-transformer.

**Comment by developer0hye** [comment]
@niujinshuchong Thanks for your reply!
